src/Button.php

```diff
@@ -4,95 +4,425 @@
+        return self::widget($constructorArguments, $config, $theme)->label($label)->type(ButtonType::LINK)->url($url);
+    }
+
+    /**
+     * Whether the button should be a reset button.
+     *
+     * @param string|stringable $value The content of the button. For default, it is 'Reset'.
+     * @param array $constructorArguments The constructor arguments.
+     * @param array $config The configuration.
+     * @param string|null $theme The theme.
+     *
+     * @return self A new instance with the button as a reset button.
+     */
+    public static function reset(
+        string|stringable $value = 'Reset',
+        array $constructorArguments = [],
+        array $config = [],
+        string|null $theme = null
+    ): self {
+        return self::widget($constructorArguments, $config, $theme)->label($value)->type(ButtonType::RESET);
+    }
+
+    /**
+     * Whether the button should be a submit button.
+     *
+     * @param string|stringable $value The content of the button. For default, it is 'Submit'.
+     * @param array $constructorArguments The constructor arguments.
+     * @param array $config The configuration.
+     * @param string|null $theme The theme.
+     *
+     * @return self A new instance with the button as a submit button.
+     */
+    public static function submit(
+        string|stringable $value = 'Submit',
+        array $constructorArguments = [],
+        array $config = [],
+        string|null $theme = null
+    ): self {
+        return self::widget($constructorArguments, $config, $theme)->label($value)->type(ButtonType::SUBMIT);
+    }
+
+    /**
+     * Sets the button to be active.
+     *
+     * @param bool $value Whether the button should be active.
+     *
+     * @return self A new instance with the button active.
+     */
+    public function active(bool $value = true): self
     {
-        if (!isset($this->options['id'])) {
-            $this->options['id'] = $this->getId();
-        }
+        $new = clone $this;
+        $new->active = $value;
 
-        if ($this->theme) {
-            $this->options['data-bs-theme'] = $this->theme;
-        }
+        return $new;
+    }
+
+    /**
+     * Adds a sets of attributes for the button component.
+     *
+     * @param array $values Attribute values indexed by attribute names. e.g. `['id' => 'my-button']`.
+     *
+     * @return self A new instance with the specified attributes added.
+     */
+    public function addAttributes(array $values): self
+    {
+        $new = clone $this;
+        $new->attributes = array_merge($this->attributes, $values);
+
+        return $new;
+    }
+
+    /**
+     * Adds a CSS class for the button group component.
+     *
+     * @param string $value The CSS class for the button component (e.g., 'test-class').
+     *
+     * @return self A new instance with the specified class value added.
+     *
+     * @link https://html.spec.whatwg.org/#classes
+     */
+    public function addClass(string $value): self
```

Comment:
Also need method that override classes (`class()`).

Reply:
What would be the use case, since all the methods are immutable, i could even overwrite the entire value.

Reply:
> You want to add `addClass()` and `class()` methods like in `html` packages

Yes

Reply:
done.